### ISSUE TITLE
Running a simple "@graph" through "dagit -f" causes "No pipelines or repositories found"

### ISSUE THREAD

**Issue by NicolasPA** [dagster-io/dagster#4449]
## Summary

Release 0.12.4

Running through Dagit a file containing only some `@op` and a `@graph` returns `dagster.core.errors.DagsterInvariantViolationError: No pipelines or repositories found` .

## Reproduction

1. Creating a simple graph in `graph_alone.py`:
```python
from dagster import op, graph


@op
def return_one():
    return 1


@graph
def my_graph():
    return_one()
````
2. Running it through Dagit: 
```shell
dagit -f graph_alone.py
```
3. Returns error: 
```shell
UserWarning: Error loading repository location graph_alone.
py:dagster.core.errors.DagsterInvariantViolationError: No pipelines or repositories found in "graph_alone".
```

This error is unexpected since it works by just replacing `@graph` with `@pipeline`:
```python
from dagster import op, pipeline


@op
def return_one():
    return 1


@pipeline
def my_graph():
    return_one()
```

Interestingly, it also works if the graph is put inside a repository:

```python
from dagster import op, graph, repository


@op
def return_one():
    return 1


@graph
def my_graph():
    return_one()


@repository
def repo():
    return [return_pipeline]
```

---
#### Message from the maintainers:
Impacted by this bug? Give it a 👍. We factor engagement into prioritization.

**Comment by sryza** [comment]
Thanks for reporting this @NicolasPA - we should definitely add support for this.  Is this gap blocking you, or just something you noticed?

**Comment by NicolasPA** [comment]
Hey @sryza, it's not blocking me because I'm using repositories, as shown in the second workaround.  I only noticed it because I wanted to debug something (op double output) with a simple example.  
Since we've just started betting on Dagster as the orchestrator for our new architecture, we wanted to start with the latest possible stage of the API, so we're taking the risk of using the experimental parts and debugging it along the way. :)
I think this issue is important to solve for the new API to quit the experimental stage.
